tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -235,18 +247,18 @@ def with_poetry(context: PipelineContext) -> Container:
     return poetry_with_cache
 
 
-def with_poetry_module(context: PipelineContext, src_path: str) -> Container:
+def with_poetry_module(context: PipelineContext, parent_dir_path: str, module_path: str) -> Container:
     """Sets up a Poetry module.
 
     Args:
         context (PipelineContext): The current test context, providing the repository directory from which the ci_credentials sources will be pulled.
     Returns:
         Container: A python environment with dependencies installed using poetry.
     """
-    poetry_exclude = ["__pycache__"] + DEFAULT_PYTHON_EXCLUDE
+    poetry_exclude = ["**/__pycache__"] + DEFAULT_PYTHON_EXCLUDE
     poetry_install_dependencies_cmd = ["poetry", "install"]
 
-    src = context.dagger_client.host().directory(src_path, exclude=poetry_exclude)
+    src = context.dagger_client.host().directory(parent_dir_path, exclude=poetry_exclude)
```

Comment:
I think we should not access FS directly in environment. I would suggest you use `get_repo_dir` on the context object. It wraps a "smart" logic to return a `Directory` object from either the Host (if local) or the Git repository (if you're running in the CI)